cmd/launcher/secure_enclave_darwin.go

```diff
@@ -46,3 +62,127 @@ func createSecureEnclaveKey() error {
 	os.Stdout.Write(secureEnclavePubDer)
 	return nil
 }
+
+func signWithSecureEnclave(signRequestB64 string) error {
+	b, err := base64.StdEncoding.DecodeString(signRequestB64)
+	if err != nil {
+		return fmt.Errorf("decoding b64 sign request: %w", err)
+	}
+
+	var signRequest secureenclavesigner.SignRequest
+	if err := json.Unmarshal(b, &signRequest); err != nil {
+		return fmt.Errorf("unmarshaling msgpack sign request: %w", err)
+	}
+
+	if err := verifySecureEnclaveChallenge(signRequest); err != nil {
+		return fmt.Errorf("verifying challenge: %w", err)
+	}
+
+	userPubKey, err := echelper.PublicB64DerToEcdsaKey(signRequest.UserPubkey)
+	if err != nil {
+		return fmt.Errorf("marshalling b64 der to public key: %w", err)
+	}
+
+	seSigner, err := secureenclave.New(userPubKey)
+	if err != nil {
+		return fmt.Errorf("creating secure enclave cmd signer: %w", err)
+	}
+
+	// tag the ends of the data to sign, this is intended to ensure that launcher wont
+	// sign arbitrary things, any party verifying the signature will need to
+	// handle these tags
+	dataToSign := []byte(fmt.Sprintf("kolide:%s:kolide", signRequest.Data))
```

Comment:
I created a PR for a doc here https://github.com/kolide/monorepo/pull/166.

We have to do it at this point because there is no way to ensure that the process which is execing launcher in the user context adds the tags.

The inability for a process to confidently detect it's parent process  is the primary driver for all this tag hoopla.